biz.aQute.bndlib/src/aQute/bnd/osgi/AnnotationHeaders.java

```diff
@@ -675,6 +676,26 @@ private String getFilter(Annotation a, Requirement annotation) {
 				}
 			}
 		}
+		else if (a.containsKey("version")) {
```

Comment:
You test `a.containsKey("version")` twice. Perhaps one test with the if/else for the `$` inside.